toolchain/semantics/testdata/tuples/builtin_implicit_empty_tuple.carbon

```diff
@@ -0,0 +1,26 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// CHECK:STDOUT: cross_reference_irs_size: 1
+// CHECK:STDOUT: functions: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: integer_literals: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: real_literals: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: strings: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: types: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: nodes: [
+// CHECK:STDOUT: ]
+// CHECK:STDOUT: node_blocks: [
+// CHECK:STDOUT:   [
+// CHECK:STDOUT:   ],
+// CHECK:STDOUT: ]
+
+// TODO test if builtin implicit empty tuple type is treated as being the same type as an explicitly-created empty tuple type.
```

Comment:
Previously I thought we'd discussed that the builtin empty tuple type should be removed. Is this the right test / TODO?

Reply:
I added this test in response to [this](https://github.com/carbon-language/carbon-lang/pull/2992#pullrequestreview-1537849360). Please let me know if it is incorrect or if I should have handled it differently. Thanks!

Reply:
@zygoloid requested there a test case that at least ensures the assignment doesn't crash. However, with the body commented out, it cannot be verifying a non-crash.

Since the implicit empty tuple is a feature of function calls, and we already have similar with function/call/i32.carbon, my suggest is to instead name this function/call/return_implicit.carbon. Test the assignment to an empty tuple, and do not comment out the code so that it does not crash.

The TODO should be that the implicit return should be a regular empty tuple.

Reply:
@zygoloid currently this case crashes however it is not because of the changes made in this PR. It is because the type id is not used [here](https://github.com/carbon-language/carbon-lang/blob/trunk/toolchain/semantics/semantics_handle_function.cpp#L20)